README.md

MMap
====================================

![Screenshot](http://holyshared.github.com/MMap/logo.png)

Map using Google Maps Api Version 3. A comprehensible map can be displayed by using the custom marker.

How to use
------------------------------------

### MMap

#### Step1 Reading of library.

Mootools and MMap are added in the head element and read. 

	#HTML
	<script type="text/javascript" src="http://maps.google.com/maps/api/js?v=3&amp;sensor=false"></script>
	<script type="text/javascript" src="js/lib/mootools-core.js"></script>
	<script type="text/javascript" src="js/lib/mmap-compressed.js"></script>

#### Step2 HTML description of main.

HTML to use MMap becomes as follows.

	#HTML
	<div id="map"></div>

#### Step3 Description of javascript.

	#JS
	(function($){
	
	window.addEvent("domready", function(){

		var map = new google.maps.Map($('map'), {
			zoom: 15,
			center: new google.maps.LatLng(35.6666870, 139.731859),
			mapTypeId: google.maps.MapTypeId.ROADMAP
		});

		var marker = new MMap.Marker.HTML({
			map: map,
			className: 'marker markerDefault',
			title: 'Marker who contains simple contents',
			content: 'HTML contents can be inserted. <br />'
				+ 'It is usual to put sentences and photographs of several lines.',
			position: new google.maps.LatLng(35.6666870, 139.731859),
			visible: true
		});

	});
	
	}(document.id));

### Online Document

* [English document](http://holyshared.github.com/MMap/docs/html/en/mmap.overlayview.html)
* [Japanese document](http://holyshared.github.com/MMap/docs/html/ja/mmap.overlayview.html)

### Online Demonstration

* [English demo](http://holyshared.github.com/MMap/marker.html)
* [Japanese demo](http://holyshared.github.com/MMap/ja/marker.html)

CHANGELOG.md

MMAP CHANGELOG
========================================

0.2.3
-------------------------------------------------------------------
* It is corrected that *_getInstance* is abolished, and *toElement* is used. 
* *MMap.Position* and *MMap.Draggable* were added. 
* It corresponded to *dragStart*, *drag*, and the *dragEnd* event of the marker. 
* The demo page bug is corrected.
* The bug of *MMap.Marker.Images* is corrected. 
* Event *onCurrentChanged* is added to MMap.Marker.Images.  
  The argument is an *index number* and an *image information*.
* Method *getCurrentImage* is added to MMap.Marker.Images.  
* MMap.Marker.Images optional *defaultIndex* is changed to *current*.

0.2.2
-------------------------------------------------------------------
* Code refactoring and bug fix.

0.2.1
-------------------------------------------------------------------
* MMap.BaseMarker to MMap.Marker.Core
* MMap.Marker to MMap.Marker.HTML
* The bug of the MMap.MarkerLoader is corrected.
* The options of MMap.MarkerLoader and the treatment of the argument are corrected.
* The active/visible method of MMap.MarkerManager is corrected.
* activeAll, activeByBounds, visibleAll, and visibleByBounds are added to MMap.MarkerManager.

0.2
-------------------------------------------------------------------
* For bind of property
* Addition of MaekerLoader
* Addition of MaekerManager
* Addition of BaseMarker
* The map of MMap is deleted.
* The specification of coordinates is united to google.maps.Latlng.
* It is corrected that set/set of MVCObject is used.

0.1
-------------------------------------------------------------------
* Comment correction

0.1b
-------------------------------------------------------------------
* Initial release

js/home.js
var EventController = new Class({

	Implements: [Events, Options],

	options: {
		"perpage": 2,
		"controller": "contentController",
		"element": "unit"
	},

	initialize: function(container, options) {
		this.setOptions(options);
		this.current = 0;
		this.container = container;
		this.setupMember();
		this.setupInterface();
		this.set(this.current);
	},

	setupMember: function() {
		this.controller = $(this.container).getElement("." + this.options.controller);
		this.calendar = $(this.container).getElement(".vcalendar");
		this.events = $(this.calendar).getElements("." + this.options.element);
	},

	setupInterface: function(){
		this.prevButton = $(this.controller).getElement(".prev");
		this.nextButton = $(this.controller).getElement(".next");
		this.prevButton.addEvent("click", this.onPrev.bind(this));
		this.nextButton.addEvent("click", this.onNext.bind(this));
	},
	
	set: function(index) {
		this.current = index;
		this.range = this.getRange();
		this.events.each(function(event, key) {
			var vevent = event.getElement(".vevent");
			(this.range.from <= key && this.range.to >= key)
			? event.setStyle("display", "")
			: event.setStyle("display", "none");
			(this.range.to == key) ? event.addClass("lastUnit") : event.removeClass("lastUnit");
			(this.current == key) ? vevent.addClass("active") : vevent.removeClass("active");
		}, this);
	},

	getRange: function() {
		var sIndex = eIndex = this.current;
		if (this.current == 0) {
			eIndex = this.current + this.options.perpage;
		} else if (this.current == this.events.length - 1) {
			sIndex = this.current - this.options.perpage;
		} else {
			sIndex = this.current - 1;
			eIndex = this.current + 1;
		}
		return {"from": sIndex, "to": eIndex};
	},

	prev: function() {
		if (this.current - 1 < 0) return false;
		this.current--;
		this.set(this.current);
	},

	next: function() {
		if (this.current + 1 >= this.events.length) return false;
		this.current++;
		this.set(this.current);
	},

	onPrev: function(event) {
		event.stop();
		this.prev();
	},

	onNext: function(event) {
		event.stop();
		this.next();
	}
});


var Home = {

	initialize: function() {
		this.map = new MMap($("map"), {
			"center": {"lat": 35.6666870, "lng": 139.731859},
			"zoom": 10,
			"mapType": "roadmap"
		});

		new MMap.Marker.Image(this.map, {
			"latlng": {"lat": 35.6666870, "lng": 139.731859},
			"src": "images/index/img_event1.jpg"
		});
		new EventController($("calendar"));
	}

};

window.addEvent("domready", Home.initialize.bind(this));